### PR TITLE
MCLOUD-6708: Add notice that only Production mode is supported on cloud

### PR DESCRIPTION
**Pull request by NadiyaS**
## Purpose of this pull request

- Added information to clarify that Magento Commerce Cloud deploys to Magento Production mode by default and that Magento developer mode is not supported.
- Updated error messages to add error for invalid mode setting

## Affected DevDocs pages

-  https://devdocs.magento.com/cloud/requirements/cloud-requirements.html
- https://devdocs.magento.com/cloud/reference/ece-tools-error-reference.html

## Links to Magento source code

https://github.com/magento/ece-tools/pull/771

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/cloud/requirements/cloud-requirements.md [hunk 1]
```diff
@@ -87,6 +87,8 @@ To get started as a developer in a {{site.data.var.ece}} project, you need to se
 
 ## Magento configurations {#cloud-req-test}
 
+Magento Commerce Cloud supports Magento which runs only in [Production mode]({{ page.baseurl }}/config-guide/bootstrap/magento-modes.html#production-mode). This mode set by default during the deployment, you should not configure it separately.
```
Comment (meker12) [inline src/cloud/requirements/cloud-requirements.md:90]:
```suggestion
{{ site.data.var.ece }} deploys Magento in [production mode]({{ page.baseurl }}/config-guide/bootstrap/magento-modes.html#production-mode) by default. Do not configure the Magento mode separately using the Magento CLI.
```

File: src/_includes/cloud/error-codes.md [hunk 2]
```diff
@@ -147,6 +147,8 @@ Warning errors indicate a problem with the Magento Commerce Cloud project config
 | 2023 | install-update:split-db | Enabling a split database will be skipped. |  |
 | 2024 | install-update:split-db | The SPLIT_DB variable is missing the configuration for split connection types. |  |
 | 2025 | install-update:split-db | Slave connection not set. |  |
+| 2026 | pre-deploy:restore-writable-dirs | Failed to restore some data generated during the build phase to the mounted directories | Check the `cloud.log` for more information. |
+| 2027 | validate-config:mage-mode-variable | Environment variable MAGE_MODE has value with not supported mode  | Remove MAGE_MODE environment variable or change its value to "production". Magento Cloud supports only "production" mode. |
```
Comment (meker12) [inline src/_includes/cloud/error-codes.md:151]: Suggest changing the error code description for 2027 in the code to:

The mode value specified in the MAGE_MODE environment variable is not supported.

Comment (NadiyaS) [reply src/_includes/cloud/error-codes.md]: Updated

File: src/_includes/cloud/error-codes.md [hunk 3]
```diff
@@ -147,6 +147,8 @@ Warning errors indicate a problem with the Magento Commerce Cloud project config
 | 2023 | install-update:split-db | Enabling a split database will be skipped. |  |
 | 2024 | install-update:split-db | The SPLIT_DB variable is missing the configuration for split connection types. |  |
 | 2025 | install-update:split-db | Slave connection not set. |  |
+| 2026 | pre-deploy:restore-writable-dirs | Failed to restore some data generated during the build phase to the mounted directories | Check the `cloud.log` for more information. |
+| 2027 | validate-config:mage-mode-variable | The mode value specified in the MAGE_MODE environment variable is not supported. | Remove MAGE_MODE environment variable or change its value to "production". Magento Cloud supports "production" mode only. |
```
Comment (meker12) [inline src/_includes/cloud/error-codes.md:151]:
```suggestion
| 2027 | validate-config:mage-mode-variable | Mode value for MAGE_MODE environment variable not supported | Remove the MAGE_MODE environment variable, or change its value to "production". Magento Cloud supports "production" mode only. |
```

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Pull request merged**